PR: https://github.com/odoo/odoo/pull/224423

From: b39b6a0b33b8585c920ce1347ac0d2ff928ac6e2
From: paan-odoo
Date: 2025-10-10 13:09:58

Breaking data model changes scores: del:9 + add:0, change matches:
-    is_order_printer = fields.Boolean('Order Printer')
-    is_posbox = fields.Boolean("PosBox")
-    is_header_or_footer = fields.Boolean("Custom Header & Footer")
-    is_closing_entry_by_product = fields.Boolean(
-    is_kiosk_mode = fields.Boolean(string="Is Kiosk Mode", default=False)
-    pos_is_order_printer = fields.Boolean(compute='_compute_pos_printer', store=True, readonly=False)
-    pos_is_header_or_footer = fields.Boolean(related='pos_config_id.is_header_or_footer', readonly=False)
-    pos_is_posbox = fields.Boolean(related='pos_config_id.is_posbox', readonly=False)
-    pos_is_closing_entry_by_product = fields.Boolean(related='pos_config_id.is_closing_entry_by_product', readonly=False)

Total Changes: 48

[IMP] pos_*: standardize boolean field names

pos_*: point_of_sale, pos_restaurant, pos_loyalty, pos_self_order,
pos_self_order_sale, l10n_in_pos

In this commit:

* Standardized the naming of boolean fields in the pos.config model for better
  consistency and readability across the codebase.

Related:
* Enterprise: https://github.com/odoo/enterprise/pull/93278
* Upgrade: https://github.com/odoo/upgrade/pull/8300

task-5008370

closes odoo/odoo#224423

Related: odoo/enterprise#93278
Related: odoo/upgrade#8300
Signed-off-by: David Monnom (moda) <moda@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -70,7 +70,7 @@ class PosConfig(models.Model):
 
     name = fields.Char(string='Point of Sale', required=True, help="An internal identification of the point of sale.")
     printer_ids = fields.Many2many('pos.printer', 'pos_config_printer_rel', 'config_id', 'printer_id', string='Order Printers')
-    is_order_printer = fields.Boolean('Order Printer')
+    use_order_printer = fields.Boolean('Order Printer')
     is_installed_account_accountant = fields.Boolean(string="Is the Full Accounting Installed",
         compute="_compute_is_installed_account_accountant")
     picking_type_id = fields.Many2one(
@@ -160,8 +160,8 @@ class PosConfig(models.Model):
     module_pos_avatax = fields.Boolean("AvaTax PoS Integration", help="Use automatic taxes mapping with Avatax in PoS")
     module_pos_discount = fields.Boolean("Global Discounts")
     module_pos_appointment = fields.Boolean("Online Booking")
-    is_posbox = fields.Boolean("PosBox")
-    is_header_or_footer = fields.Boolean("Custom Header & Footer")
+    use_posbox = fields.Boolean("PosBox")
+    use_header_or_footer = fields.Boolean("Custom Header & Footer")
     module_pos_hr = fields.Boolean(help="Show employee login screen")
     amount_authorized_diff = fields.Float('Amount Authorized Difference',
         help="This field depicts the maximum difference allowed between the ending balance and the theoretical cash when "
@@ -194,7 +194,7 @@ class PosConfig(models.Model):
     show_category_images = fields.Boolean(string="Show Category Images", help="Show category images in the Point of Sale interface.", default=True)
     note_ids = fields.Many2many('pos.note', string='Note Models', help='The predefined notes of this point of sale.')
     module_pos_sms = fields.Boolean(string="SMS Enabled", help="Activate SMS feature for point_of_sale")
-    is_closing_entry_by_product = fields.Boolean(
+    use_closing_entry_by_product = fields.Boolean(
         string='Closing Entry by product',
         help="Display the breakdown of sales lines by product in the automatically generated closing entry.")
     order_edit_tracking = fields.Boolean(string="Track orders edits", help="Store edited orders in the backend", default=False)
@@ -559,7 +559,7 @@ class PosConfig(models.Model):
                     raise ValidationError(_("You cannot share open orders with configuration that does not use the same currency."))
 
     def _check_header_footer(self, values):
-        if not self.env.is_admin() and {'is_header_or_footer', 'receipt_header', 'receipt_footer'} & values.keys():
+        if not self.env.is_admin() and {'use_header_or_footer', 'receipt_header', 'receipt_footer'} & values.keys():
             raise AccessError(_('Only administrators can edit receipt headers and footers'))
 
     def _check_company_has_fiscal_country(self):
@@ -637,7 +637,7 @@ class PosConfig(models.Model):
     def _update_preparation_printers_menuitem_visibility(self):
         prepa_printers_menuitem = self.sudo().env.ref('point_of_sale.menu_pos_preparation_printer', raise_if_not_found=False)
         if prepa_printers_menuitem:
-            prepa_printers_menuitem.active = self.sudo().env['pos.config'].search_count([('is_order_printer', '=', True)], limit=1) > 0
+            prepa_printers_menuitem.active = self.sudo().env['pos.config'].search_count([('use_order_printer', '=', True)], limit=1) > 0
 
     @api.depends('use_pricelist', 'pricelist_id', 'available_pricelist_ids', 'payment_method_ids', 'limit_categories',
         'iface_available_categ_ids', 'module_pos_hr', 'module_pos_discount', 'iface_tipproduct', 'default_preset_id', 'module_pos_appointment')
@@ -647,7 +647,7 @@ class PosConfig(models.Model):
     def write(self, vals):
         self._check_header_footer(vals)
         self._reset_default_on_vals(vals)
-        if ('is_order_printer' in vals and not vals['is_order_printer']):
+        if ('use_order_printer' in vals and not vals['use_order_printer']):
             vals['printer_ids'] = [fields.Command.clear()]
 
         bypass_payment_method_ids_forbidden_change = self.env.context.get('bypass_payment_method_ids_forbidden_change', False)
@@ -679,7 +679,7 @@ class PosConfig(models.Model):
         self.sudo()._set_fiscal_position()
         self.sudo()._check_modules_to_install()
         self.sudo()._check_groups_implied()
-        if 'is_order_printer' in vals:
+        if 'use_order_printer' in vals:
             self._update_preparation_printers_menuitem_visibility()
         return result
 
@@ -1267,7 +1267,7 @@ class PosConfig(models.Model):
             param_model.set_param("point_of_sale.limited_customer_count", DEFAULT_LIMIT_LOAD_PARTNER)
 
     def _is_quantities_set(self):
-        return self.is_closing_entry_by_product
+        return self.use_closing_entry_by_product
 
     @api.onchange("epson_printer_ip")
     def _onchange_epson_printer_ip(self):

--- a/addons/point_of_sale/models/pos_session.py
+++ b/addons/point_of_sale/models/pos_session.py
@@ -1380,7 +1380,7 @@ class PosSession(models.Model):
             # for taxes
             'tax_ids': tuple(base_line['record'].tax_ids_after_fiscal_position.flatten_taxes_hierarchy().ids),
             'base_tag_ids': tuple(base_line['tax_tag_ids'].ids),
-            'product_id': base_line['product_id'].id if self.config_id.is_closing_entry_by_product else False,
+            'product_id': base_line['product_id'].id if self.config_id.use_closing_entry_by_product else False,
         }
 
     def _get_sale_vals(self, key, sale_vals):

--- a/addons/point_of_sale/models/res_config_settings.py
+++ b/addons/point_of_sale/models/res_config_settings.py
@@ -41,7 +41,7 @@ class ResConfigSettings(models.TransientModel):
     update_stock_quantities = fields.Selection(related="company_id.point_of_sale_update_stock_quantities", readonly=False)
     account_default_pos_receivable_account_id = fields.Many2one(string='Default Account Receivable (PoS)', related='company_id.account_default_pos_receivable_account_id', readonly=False, check_company=True)
     barcode_nomenclature_id = fields.Many2one('barcode.nomenclature', related='company_id.nomenclature_id', readonly=False)
-    is_kiosk_mode = fields.Boolean(string="Is Kiosk Mode", default=False)
+    use_kiosk_mode = fields.Boolean(string="Is Kiosk Mode", default=False)
     pos_customer_display_bg_img = fields.Image(related='pos_config_id.customer_display_bg_img', readonly=False)
     pos_customer_display_bg_img_name = fields.Char(related='pos_config_id.customer_display_bg_img_name', readonly=False)
 
@@ -54,7 +54,7 @@ class ResConfigSettings(models.TransientModel):
     pos_module_pos_restaurant = fields.Boolean(related='pos_config_id.module_pos_restaurant', readonly=False)
     pos_module_pos_appointment = fields.Boolean(related="pos_config_id.module_pos_appointment", readonly=False)
     pos_module_pos_avatax = fields.Boolean(related='pos_config_id.module_pos_avatax', readonly=False)
-    pos_is_order_printer = fields.Boolean(compute='_compute_pos_printer', store=True, readonly=False)
+    pos_use_order_printer = fields.Boolean(compute='_compute_pos_printer', store=True, readonly=False)
     pos_printer_ids = fields.Many2many(related='pos_config_id.printer_ids', readonly=False)
 
     pos_allowed_pricelist_ids = fields.Many2many('product.pricelist', compute='_compute_pos_allowed_pricelist_ids')
@@ -79,9 +79,9 @@ class ResConfigSettings(models.TransientModel):
     pos_iface_tax_included = fields.Selection(related='pos_config_id.iface_tax_included', readonly=False)
     pos_iface_tipproduct = fields.Boolean(related='pos_config_id.iface_tipproduct', readonly=False)
     pos_invoice_journal_id = fields.Many2one(related='pos_config_id.invoice_journal_id', readonly=False)
-    pos_is_header_or_footer = fields.Boolean(related='pos_config_id.is_header_or_footer', readonly=False)
+    pos_use_header_or_footer = fields.Boolean(related='pos_config_id.use_header_or_footer', readonly=False)
     pos_is_margins_costs_accessible_to_every_user = fields.Boolean(related='pos_config_id.is_margins_costs_accessible_to_every_user', readonly=False)
-    pos_is_posbox = fields.Boolean(related='pos_config_id.is_posbox', readonly=False)
+    pos_use_posbox = fields.Boolean(related='pos_config_id.use_posbox', readonly=False)
     pos_journal_id = fields.Many2one(related='pos_config_id.journal_id', readonly=False)
     pos_limit_categories = fields.Boolean(related='pos_config_id.limit_categories', readonly=False)
     pos_manual_discount = fields.Boolean(related='pos_config_id.manual_discount', readonly=False)
@@ -113,7 +113,7 @@ class ResConfigSettings(models.TransientModel):
     point_of_sale_ticket_portal_url_display_mode = fields.Selection(related='company_id.point_of_sale_ticket_portal_url_display_mode', readonly=False, required=True)
     pos_note_ids = fields.Many2many(related='pos_config_id.note_ids', readonly=False)
     pos_module_pos_sms = fields.Boolean(related="pos_config_id.module_pos_sms", readonly=False)
-    pos_is_closing_entry_by_product = fields.Boolean(related='pos_config_id.is_closing_entry_by_product', readonly=False)
+    pos_use_closing_entry_by_product = fields.Boolean(related='pos_config_id.use_closing_entry_by_product', readonly=False)
     pos_order_edit_tracking = fields.Boolean(related="pos_config_id.order_edit_tracking", readonly=False)
     pos_basic_receipt = fields.Boolean(related='pos_config_id.basic_receipt', readonly=False)
     pos_fallback_nomenclature_id = fields.Many2one(related='pos_config_id.fallback_nomenclature_id', domain="[('id', '!=', barcode_nomenclature_id)]", readonly=False)
@@ -240,7 +240,7 @@ class ResConfigSettings(models.TransientModel):
     def _compute_pos_printer(self):
         for res_config in self:
             res_config.update({
-                'pos_is_order_printer': res_config.pos_config_id.is_order_printer,
+                'pos_use_order_printer': res_config.pos_config_id.use_order_printer,
             })
 
     @api.depends('pos_limit_categories', 'pos_config_id')
@@ -267,10 +267,10 @@ class ResConfigSettings(models.TransientModel):
             else:
                 res_config.pos_iface_cashdrawer = False
 
-    @api.depends('pos_is_header_or_footer', 'pos_config_id')
+    @api.depends('pos_use_header_or_footer', 'pos_config_id')
     def _compute_pos_receipt_header_footer(self):
         for res_config in self:
-            if res_config.pos_is_header_or_footer:
+            if res_config.pos_use_header_or_footer:
                 res_config.pos_receipt_header = res_config.pos_config_id.receipt_header
                 res_config.pos_receipt_footer = res_config.pos_config_id.receipt_footer
             else:
@@ -322,26 +322,26 @@ class ResConfigSettings(models.TransientModel):
             else:
                 res_config.pos_allowed_pricelist_ids = self.env['product.pricelist'].search([]).ids
 
-    @api.depends('pos_is_posbox', 'pos_config_id')
+    @api.depends('pos_use_posbox', 'pos_config_id')
     def _compute_pos_iface_print_via_proxy(self):
         for res_config in self:
-            if not res_config.pos_is_posbox:
+            if not res_config.pos_use_posbox:
                 res_config.pos_iface_print_via_proxy = False
             else:
                 res_config.pos_iface_print_via_proxy = res_config.pos_config_id.iface_print_via_proxy
 
-    @api.depends('pos_is_posbox', 'pos_config_id')
+    @api.depends('pos_use_posbox', 'pos_config_id')
     def _compute_pos_iface_scan_via_proxy(self):
         for res_config in self:
-            if not res_config.pos_is_posbox:
+            if not res_config.pos_use_posbox:
                 res_config.pos_iface_scan_via_proxy = False
             else:
                 res_config.pos_iface_scan_via_proxy = res_config.pos_config_id.iface_scan_via_proxy
 
-    @api.depends('pos_is_posbox', 'pos_config_id')
+    @api.depends('pos_use_posbox', 'pos_config_id')
     def _compute_pos_iface_electronic_scale(self):
         for res_config in self:
-            if not res_config.pos_is_posbox:
+            if not res_config.pos_use_posbox:
                 res_config.pos_iface_electronic_scale = False
             else:
                 res_config.pos_iface_electronic_scale = res_config.pos_config_id.iface_electronic_scale
